### PR TITLE
Install xdebug in php80-fpm

### PR DESCRIPTION
**Pull request by janmartenjongerius**
Install the xdebug extension in the PHP 8 FPM image.

The corresponding xdebug settings have been upgraded. The configuration
`xdebug.discover_client_host` has been set to `true` to allow flexible
host configuration for IDEs through HTTP headers.

/cc @JeroenBoersma

### CODE REVIEW

File: php80/fpm/conf/php.ini [hunk 1]
```diff
@@ -5,7 +5,7 @@ memory_limit = 512M
 log_errors = 1
 error_reporting = E_ALL
 
-xdebug.remote_enable = 1
-xdebug.remote_host = 172.17.0.1
+xdebug.client_host = 172.17.0.1
+xdebug.discover_client_host = true
```
Comment (janmartenjongerius) [inline php80/fpm/conf/php.ini:9]: See https://xdebug.org/docs/all_settings#discover_client_host

File: php80/fpm/conf/php.ini [hunk 2]
```diff
@@ -5,7 +5,7 @@ memory_limit = 512M
 log_errors = 1
 error_reporting = E_ALL
 
-xdebug.remote_enable = 1
```
Comment (janmartenjongerius) [inline php80/fpm/conf/php.ini:8]: This setting has been replaced by `xdebug.mode` with value `debug`.
- https://xdebug.org/docs/upgrade_guide

File: php80/fpm/Dockerfile [hunk 3]
```diff
@@ -35,6 +35,11 @@ RUN docker-php-ext-install bcmath mysqli pdo_mysql soap zip intl opcache xsl pcn
     && docker-php-ext-configure gd --with-freetype --with-jpeg --with-webp \
     && docker-php-ext-install gd
 
+# Install xdebug
+RUN pecl install xdebug \
+    && docker-php-ext-enable xdebug \
+    && rm -rf /tmp/pear
```
Comment (JeroenBoersma) [inline php80/fpm/Dockerfile:52]: pecl was removed from php 8, was t re-added?

**Review by JeroenBoersma — CHANGES_REQUESTED** [review]

**Comment by johmanx10** [comment]
When I run a docker build and then a docker run on the built image with php
-v, it shows xdebug installed.

Leave it to the author of xdebug to fix this for us:
https://github.com/docker-library/php/issues/1087

File: php80/fpm/Dockerfile (hunk 3 above)
Comment (janmartenjongerius) [reply php80/fpm/Dockerfile:52]: When I run a `docker build` and then a `docker run` on the built image with `php -v`, it shows xdebug installed.

image

Leave it to the author of xdebug to fix this for us: docker-library/php#1087

Comment (janmartenjongerius) [reply php80/fpm/Dockerfile:52]: So, officially it wasn't removed, just not enabled. It explicitly required `--with-pear` it seems, which has been added to the official PHP images.

**Comment by janmartenjongerius** [comment]
It dawns on me that earlier versions also use xdebug 3. I had a look and my proposed settings deviate from what's already there.

Differences are:

- The setting `xdebug.discover_client_host` is added
- The setting `xdebug.mode` is set to `coverage,debug,develop` instead of `develop,debug` in other versions.
  > Adding `coverage` allows things like running PHPUnit with code coverage enabled.

Would you prefer I either:

- Make PHP 8 config the same as other version
- Update the config of other versions to be like my proposal gor PHP 8

Basically, the question is whether you agree with the changes I propose.

/cc @JeroenBoersma

**Review by JeroenBoersma — APPROVED** [review]
